src/ucp/tag/tag_rndv.c

```diff
@@ -61,27 +59,27 @@ ucs_status_t ucp_tag_rndv_process_rts(ucp_worker_h worker,
      * (and therefore keep fast search by ucp_tag_unexp_search())
      */
     status = ucp_recv_desc_init(worker, rts_hdr, length, sizeof(*rdesc_hdr),
-                                tl_flags, sizeof(*rts_hdr) + sizeof(*rdesc_hdr),
-                                UCP_RECV_DESC_FLAG_RNDV,
-                                sizeof(*rdesc_hdr), &rdesc);
+                                tl_flags, sizeof(*rts_hdr),
+                                UCP_RECV_DESC_FLAG_RNDV, sizeof(*rdesc_hdr),
+                                &rdesc);
     if (!UCS_STATUS_IS_ERR(status)) {
         rdesc_hdr  = (ucp_tag_t*)(rdesc + 1);
-        *rdesc_hdr = rts_hdr->tag.tag;
-        ucp_tag_unexp_recv(&worker->tm, rdesc, rts_hdr->tag.tag);
+        *rdesc_hdr = ucp_tag_hdr_from_rts(rts_hdr)->tag;
+        ucp_tag_unexp_recv(&worker->tm, rdesc,
+                           ucp_tag_hdr_from_rts(rts_hdr)->tag);
     }
```

Comment:
there is not need to insert additional tag before the rts now. Also please remove the comment on lines 58-60

Reply:
good catch
done